pkg/plugins/policies/meshhttproute/api/v1alpha1/compare.go

```diff
@@ -0,0 +1,150 @@
+package v1alpha1
+
+import (
+	"slices"
+
+	common_api "github.com/kumahq/kuma/api/common/v1alpha1"
+	"github.com/kumahq/kuma/pkg/util/pointer"
+)
+
+func comparePath(a *PathMatch, b *PathMatch) int {
+	switch {
+	case a != nil && b == nil:
+		return -1
+	case a == nil && b != nil:
+		return 1
+	case a == nil && b == nil:
+		return 0
+	}
+
+	switch {
+	case a.Type == Exact && b.Type == Exact:
+	case a.Type == Exact && b.Type != Exact:
+		return -1
+	case a.Type != Exact && b.Type == Exact:
+		return 1
+	}
+
+	switch {
+	case a.Type == PathPrefix && b.Type != PathPrefix:
+		return -1
+	case a.Type != PathPrefix && b.Type == PathPrefix:
+		return 1
+	case a.Type == PathPrefix && b.Type == PathPrefix:
+		switch {
+		case len(a.Value) > len(b.Value):
```

Comment:
Any reason not to use `cmp.Compare` ?

Reply:
Wouldn't that just do a lexicographic compare of the strings?

Reply:
Yeah this could be something like

```
if lenCmp := cmp.Compare(len(a.Value), len(b.Value)); lenCmp != 0 {
  return lenCmp
}
```

I guess